openid-4-verifiable-presentations-1_0.md

```diff
@@ -548,6 +548,8 @@ Note: The Verifier's component providing the user interface (Frontend) and the V
 
 Note: If the Client Identifier scheme `redirect_uri` is used in conjunction with the Response Mode `direct_post`, and the `response_uri` parameter is present, the `client_id` value MUST be equal to the `response_uri` value.
 
+Note: If the Client Identifier scheme `x509_san_dns` or `x509_san_uri` is used in conjunction with the Response Mode `direct_post` or `direct_post.jwt`, and the `response_uri` parameter is present, the `response_uri` MUST be used instead of the `redirect_uri` to compute the `client_id` value when applicable.
```

Comment:
I'm not sure 'compute' is the best word to use here. Doesn't the wallet explicitly state the `client_id` and hence it is not necessary to 'compute' a `client_id`?

The "and the `response_uri` parameter is present" part is unnecessary as `response_uri` is mandatory when using direct_post / direct_post.jwt.

I think we should avoid mentioning `direct_post.jwt` here as direct_post.jwt builds on top of direct_post. (If we are going to mention direct_post.jwt in the direct_post section we need to add "or direct_post.jwt" in most of the preceding 6 paragraphs.)

```suggestion
Note: If the Client Identifier scheme `x509_san_dns` or `x509_san_uri` is used in conjunction with the Response Mode `direct_post`, the `response_uri` MUST be a value that the client would be permitted to use as a `redirect_uri` when following the rules defined in (#client_metadata_management).
```

Reply:
Although as an alternative how about removing both these notes and simply saying something that is generically applicable to all client id schemes:

> `response_uri` MUST be a value that the client would be permitted to use as a `redirect_uri` when following the rules defined in (#client_metadata_management).

Reply:
That sounds reasonable.

Reply:
> Although as an alternative how about removing both these notes and simply saying something that is generically applicable to all client id schemes - I can't imagine any case where response_uri would have a different restriction than redirect_uri does?
> 
> i.e.:
> 
> > `response_uri` MUST be a value that the client would be permitted to use as a `redirect_uri` when following the rules defined in (#client_metadata_management).

I think the above proposal is also not 100% accurate. What about?

> The `response_uri` value MUST be used instead of the `redirect_uri` Authorization Request parameter when following the rules defined in (#client_metadata_management).

IMO, this makes it clear that `redirect_uri` in the Response URI endpoint response can still be different from the `response_uri` Authz Request parameter but the `response_uri` Authz Request parameter has to be used for the `client_id` when applicable as defined in the client ID schemes section.

Reply:
I re-checked the spec. There is no need to state "if the response_uri is present" as it MUST be present in case of response mode "direct_post".

Reply:
You are right. I updated the PR. Please check again.